### PR TITLE
MINOR: Streams upgrade guide section for newly merged KIPs

### PR DESCRIPTION
**Pull request by guozhangwang**
@shawnsnguyen for KIP-393.
@shunge for KIP-376
And also the state transition change.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
@mjsax @bbejeck @vvcephei

**Review by bbejeck — APPROVED** [review]
Thanks @guozhangwang LGTM

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -68,6 +68,23 @@ <h1>Upgrade Guide and API Changes</h1>
         More details about the new config <code>StreamsConfig#TOPOLOGY_OPTIMIZATION</code> can be found in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-295%3A+Add+Streams+Configuration+Allowing+for+Optional+Topology+Optimization">KIP-295</a>.
     </p>
 
+    <h3><a id="streams_api_changes_220" href="#streams_api_changes_220">Streams API changes in 2.2.0</a></h3>
+    <p>
+        We've simplified the <code>KafkaStreams#state</code> transition diagram during the starting up phase a bit in 2.2.0: in older versions the state will transit from <code>CREATED</code> to <code>RUNNING</code>, and then to <code>REBALANCING</code> to get the first
+        stream task assignment, and then back to <code>RUNNING</code>; starting in 2.2.0 it will transit from <code>CREATED</code> directly to <code>REBALANCING</code> and then to <code>RUNNING</code>.
+        If you have registered a <code>StateListener</code> that captures state transition events, you may need to adjust your listener implementation accordingly for this simplification (in practice, your listener logic should very likely to not be affected at all).
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: `likely to not` (remove `to`)

**Review by mjsax — COMMENTED** [review]
Should we add a paragraph for KIP-331, too? I guess for KIP-386, we don't need to do anything?

**Comment by guozhangwang** [comment]
@mjsax thanks for the reminder.

KIP-331's PR (https://github.com/apache/kafka/pull/5348) has not been merged yet. 
KIP-386's corresponding content should be in docs/upgrade.html.

File: docs/streams/upgrade-guide.html (hunk 1 above)
Comment (guozhangwang) [reply docs/streams/upgrade-guide.html]: Ack.

**Review by mjsax — COMMENTED** [review]
Ack. LGTM. Feel free to merge after fixing the typo.

**Pull request merged**